src/main/java/hudson/plugins/git/util/DefaultBuildChooser.java

```diff
@@ -121,7 +127,41 @@ public DefaultBuildChooser() {
             }
         }
         
-        return revisions;
+        return sortByCriticalBranches(revisions, listener);
+    }
+    
+    private Collection<Revision> sortByCriticalBranches(Collection<Revision> revisions, TaskListener listener) {
+        Comparator<Revision> revisionComparator = new Comparator<Revision>() {
+			public int compare(Revision r1, Revision r2) {
+				return branchWithCritical(r2.getBranches()) - branchWithCritical(r1.getBranches());
+			}
+			
+			private int branchWithCritical(Collection<Branch> branches) {
+				for (Branch branch : branches) {
+					if (branch.getName().contains("CRITICAL")) {
```

Comment:
If we have to deploy this on monday then that's fine but ideally this would be a configurable parameter. We can open another issue for that change.